tests/v3/test_store/test_remote.py

```diff
@@ -81,31 +95,51 @@ async def test_basic():
 class TestRemoteStoreS3(StoreTests[RemoteStore]):
     store_cls = RemoteStore
 
-    @pytest.fixture(scope="function")
-    def store_kwargs(self) -> dict[str, str | bool]:
-        return {
-            "mode": "w",
-            "endpoint_url": endpoint_uri,
-            "anon": False,
-            "url": f"s3://{test_bucket_name}",
-        }
+    @pytest.fixture(scope="function", params=("use_upath", "use_str"))
+    def store_kwargs(self, request) -> dict[str, str | bool]:
+        url = f"s3://{test_bucket_name}"
+        anon = False
+        mode = "w"
+        if request.param == "use_upath":
+            return {"mode": mode, "url": UPath(url, endpoint_url=endpoint_url, anon=anon)}
+        elif request.param == "use_str":
+            return {"url": url, "mode": mode, "anon": anon, "endpoint_url": endpoint_url}
+        else:
+            raise AssertionError
 
     @pytest.fixture(scope="function")
     def store(self, store_kwargs: dict[str, str | bool]) -> RemoteStore:
-        self._fs, _ = fsspec.url_to_fs(asynchronous=False, **store_kwargs)
-        out = self.store_cls(asynchronous=True, **store_kwargs)
+        url = store_kwargs["url"]
+        mode = store_kwargs["mode"]
+        if isinstance(url, UPath):
+            out = self.store_cls(url=url, mode=mode)
+        else:
+            endpoint_url = store_kwargs["endpoint_url"]
+            out = self.store_cls(url=url, asynchronous=True, mode=mode, endpoint_url=endpoint_url)
         return out
 
     def get(self, store: RemoteStore, key: str) -> Buffer:
-        return Buffer.from_bytes(self._fs.cat(f"{store.path}/{key}"))
+        #  make a new, synchronous instance of the filesystem because this test is run in sync code
+        fs, _ = fsspec.url_to_fs(
```

Comment:
Seems like going around in a circle here: str->UPath(which contains fs)->str->fs. Shouldn't it be the same as `store.fs`?

Reply:
unless I'm misunderstanding your concern, the issue with `store._fs` is that it's async, but `StoreTests.get` and `StoreTests.set` are sync, and so using `store._fs` in those methods doesn't work

Reply:
Right of course, sorry.